dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/support/MockInvoker.java

```diff
@@ -177,12 +178,27 @@ public static Throwable getThrowable(String throwstr) {
 
     @SuppressWarnings("unchecked")
     public static Object getMockObject(String mockService, Class serviceType) {
-        if (ConfigUtils.isDefault(mockService)) {
+        boolean isDefault = ConfigUtils.isDefault(mockService);
+        if (isDefault) {
             mockService = serviceType.getName() + "Mock";
         }
 
-        Class<?> mockClass = ReflectUtils.forName(mockService);
-        if (!serviceType.isAssignableFrom(mockClass)) {
+        Class<?> mockClass;
+        try {
+            mockClass = ReflectUtils.forName(mockService);
+        } catch (Exception e) {
+            if (!isDefault) {// does not check Spring bean if it is default config.
+                ExtensionFactory extensionFactory =
+                        ExtensionLoader.getExtensionLoader(ExtensionFactory.class).getAdaptiveExtension();
+                Object obj = extensionFactory.getExtension(serviceType, mockService);
+                if (obj != null) {
+                    return obj;
+                }
+            }
+            throw new IllegalStateException("Did not find mock class or instance " + mockService + ", please check" +
+                    " if there's mock class or instance implementing interface " + serviceType.getName(), e);

```

Comment:
` please check" + " if there's mock class or instance `   put it on line would be better